src/components/InterestForm/index.tsx

```diff
@@ -1,73 +1,101 @@
-import { P, H3 } from '@/styles/text';
-import COLORS from '@/styles/colors';
-import DateInput from '../DateInput';
-import Button from '../Button';
-import {
-  FormContainer,
-  FormTextArea,
-  FormQuestion,
-  RadioGroup,
-  Radio,
-  RadioInput,
-  FormFooter,
-  FormWarning,
-  ErrorText,
-  RadioLabel,
-} from './styles';
+import { useEffect, useState } from 'react';
+import { EmptySpace, FormContainer, FormFooter, FormWarning } from './styles';
 
-const radioOptions = [
-  'Attorney',
-  'Interpreter',
-  'Either Attorney or Interpreter',
-];
+interface Responses {
+  start_date?: Date;
+  needs_interpreter?: boolean;
+  interest_reason: string;
+}
 
-export default function InterestForm({ caseData }: { caseData: CaseListing }) {
+export default function InterestForm({
+  listingData,
+  interpretation = false,
+}: {
+  listingData: Listing;
+  interpretation?: boolean;
+}) {
   const auth = useAuth();
   const [reason, setReason] = useState<string>('');
-  const [rolesInterested, setRolesInterested] = useState<string>('');
+  const [needsInterpreter, setNeedsInterpreter] = useState<string>('');
   const [startDate, setStartDate] = useState<string>('');
   const [submitted, setSubmitted] = useState(false);
   const [missingInfo, setMissingInfo] = useState(false);
 
   useEffect(() => {
     // Reset form fields when caseData changes
     setReason('');
+    setNeedsInterpreter('');
     setStartDate('');
-    setRolesInterested('');
     setSubmitted(false);
     setMissingInfo(false);
-  }, [caseData]);
+  }, [listingData]);
 
   const handleInsert = async () => {
-    if (startDate === '' || rolesInterested === '') {
+    // Error handling, check if required fields are unfilled.
+    if (
+      // CASE listing requires: startDate and needsInterpreter
+      (listingData.listing_type === 'CASE' &&
+        !interpretation &&
+        (startDate === '' || needsInterpreter === '')) ||
+      // CASE_INT and INT listings require: startDate
+      (((listingData.listing_type === 'CASE' && interpretation) ||
+        listingData.listing_type === 'INT') &&
+        startDate === '') ||
+      // LCA and DOC listings require: reason
+      ((listingData.listing_type === 'LCA' ||
+        listingData.listing_type === 'DOC') &&
+        reason === '')
+    ) {
       setMissingInfo(true);
       return;
     }
-    if (isValidDate(startDate)) {
+    const responses: Responses = { interest_reason: '' };
+    if (
+      isValidDate(startDate) ||
+      listingData.listing_type === 'DOC' ||
+      listingData.listing_type === 'LCA'
+    ) {
       if (auth && auth.userId) {
+        if (
+          listingData.listing_type === 'CASE' ||
+          listingData.listing_type === 'INT'
+        ) {
+          responses.start_date = new Date(startDate);
+          if (listingData.listing_type === 'CASE' && !interpretation) {
+            responses.needs_interpreter = needsInterpreter === 'Yes';
+          }
```

Comment:
let's add comments for these logic as well

Reply:
added !